server/src/main/resources/config/play.conf

```diff
@@ -3,3 +3,51 @@ play.akka.actor-system = ${portcode}"-drt-actor-system"
 play.http.parser.maxMemoryBuffer = 100m
 
 play.server.netty.maxHeaderSize = 32768
+
+play.filters.csp.nonce.enabled = true
+
+play.filters.hosts {
+  allowed = [".", "0.0.0.0:9000"]
+}
+
+play.filters.csp.nonce {
+  # Use nonce value (generated and passed in through request attribute)
+  enabled = true
+
+  # Pattern to use to replace with nonce
+  pattern = "%CSP_NONCE_PATTERN%"
+
+  # Add the nonce to "X-Content-Security-Policy-Nonce" header.  This is useful for debugging.
+  header = false
+}
+
+play.filters.headers.frameOptions = SAMEORIGIN
+play.filters.csp.reportOnly = true
+
+play.filters.csrf.contentType.whiteList += "application/csp-report"
+
+play.filters.enabled += play.filters.csp.CSPFilter
+
+play.filters.csp {
+  hashes += {
+    algorithm = "sha256"
+    hash = "RpniQm4B6bHP0cNtv7w1p6pVcgpm5B/eu1DNEYyMFXc="
+    pattern = "%CSP_MYSCRIPT_HASH%"
+  }
+  style-src = "%CSP_MYSCRIPT_HASH%"
+}
+
+play.filters.csp {
+  directives {
+    report-to = "http://localhost:9000/report-to"
+    report-uri = ${play.filters.csp.directives.report-to}
+    base-uri = "'self'"
+    object-src = "'self'"
+    default-src = "'self'"
+    style-src = "'self' cdnjs.cloudflare.com 'unsafe-inline'"
+    font-src = "'self' cdnjs.cloudflare.com"
+    img-src = "'self' https://*.googletagmanager.com www.googletagmanager.com https://*.google-analytics.com www.google-analytics.com"
```

Comment:
For my own understanding - why is the google stuff with `img-src`? I'd expect `img-src` to have images with it

Reply:
img-src allows sources to load images on the web page. So you correct as we don't images from googletagmanger or analytics we can remove them .